.github/workflows/dev-pr.yml

```diff
@@ -0,0 +1,101 @@
+name: Dev PR
+
+on:
+  pull_request:
+    branches:
+      - dev
+
+env:
+  CARGO_TERM_COLOR: always
+  ARCH_TYPE: sm_70
+  LD_LIBRARY_PATH: $GITHUB_WORKSPACE/goicicle
+
+jobs:
+  build-rust-linux:
+    name: Build Rust on Linux
+    runs-on: [self-hosted, Linux, X64, icicle]
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v3
+    - name: Build Rust
+      run: cargo build --release --verbose
+
+  test-rust-linux:
+    name: Test Rust on Linux
+    needs: build-rust-linux
+    runs-on: [self-hosted, Linux, X64, icicle]
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v3
+    - name: Run Rust Tests
+      run: cargo test --release --verbose -- --test-threads=1
+
+  formatting-rust:
+    name: Check Rust Code Formatting
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Check rustfmt
+      run: if [[ $(cargo fmt --check) ]]; then echo "Please run cargo fmt"; exit 1; fi
+    # - name: Check clippy
+    #   run: cargo clippy --no-deps --all-features --all-targets
+
+  build-rust-windows:
+    name: Build Rust on Windows
+    runs-on: windows-2022
+    steps:     
+    - name: Checkout Repo
+      uses: actions/checkout@v3
+    - name: Download and Install Cuda
+      uses: Jimver/cuda-toolkit@v0.2.11
+      with:
+        cuda: '12.0.0'
+        method: 'network'
+        # https://docs.nvidia.com/cuda/archive/12.0.0/cuda-installation-guide-microsoft-windows/index.html
+        sub-packages: '["cudart", "nvcc", "thrust"]'
+    - name: Build Rust Targets
+      run: cargo build --release --verbose
+
+  test-golang-linux:
+    name: Test Golang on Linux
+    runs-on: [self-hosted, Linux, X64, icicle]
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v3
+    - name: Build CUDA libs
+      run: make libbn254.so
+      working-directory: ./goicicle
+    - name: Run Golang Tests
+      run: |
+        export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$(pwd)/goicicle
+        go test ./goicicle/curves/bn254 -count=1
```

Comment:
We probably need some more general testing behavior, TODO

Reply:
What do you mean by "general"?

ideally, we would use `go test ./... -count=1` which would run all tests that we have but 377 and 381 aren't working right now.

Reply:
I mean we can run all tests with something like `go test -run ''` or specify tests to run on dev or main `go test -run Dev`

Reply:
Yea, agreed. We should use `go test ./... -count=1` but it will fail on bls12377 and bls12388 tests which have TODOs